File: Ld1_181rdb178.py

import math
import sys

def valueA(num):
    if num>0:
        a=num*num+math.log(num)
        return a;
    else: 
        print("Ar ievadīto skaitli funkciju aprēķināt nevar.")
        mainTask();
        #sys.exit() nevis mainTask() ja grib pārtraukt programmu
    
def valueB(num):
    if (num+num*num-10)>0:
        b=math.sqrt(num+num*num-10)
        return b;
    else:
        print("Ar ievadīto skaitli funkciju aprēķināt nevar.")
        mainTask();
        #sys.exit() nevis mainTask() ja grib pārtraukt programmu
    
def funkcija1(num1,num2):
    y1=num1/2 +(num1+num2)/2
    return y1;
    
def funkcija2(num1,num2):
    y2=(num1*num1 +num2*num2)/(num1*num2)
    return y2;
    
def comparison(valA,valB):
    if (valA/10)<valB :
        print("Rezultāts funkcijai Y: ", "%.3f" %funkcija1(valA,valB))
    else:
        print("Rezultāts funkcijai Y: ", "%.3f" %funkcija2(valA,valB))

def ievade():
    while True:
        user_input=input("Ievadiet skaitli X:")
        try:
            valueX=float(user_input)
            return valueX
            #print("viss ok")
            break
        except ValueError:
            print("Kļūda ievadē")

print("Jolanta Bērziņa 181RDB178  8.grupa")

def mainTask():
    valX=ievade()
    a=valueA(valX)
    b=valueB(valX)
    comparison(a,b)
    
    answer=input("Vai turpināt darbu ar programmu?[y/n]")
    if (answer=="y" or answer=="Y"):
        mainTask()
    else:
        sys.exit()

mainTask()

File: Ld1_181rdb178.cpp


#include <iostream>
#include <math.h>
#include <stdio.h>
#include <iomanip>
#include <cmath>

using namespace std;

float valueA (float x){
    float valA = x*x+ log(x);
    return valA;
}

float valueB (float x){
    float valB =sqrt(x+x*x-10);
    return valB;
}

float function1 (float a, float b) {
    float Y1  = a/2 +(a+b)/2;
    return Y1;
}

float function2 (float a, float b){
    float Y2 = (a*a +b*b)/(a*b);
    return Y2;
}

void comparison (float a,float b){
    if ((a/10)<b) {
        cout<<"Rezultāts funkcijai Y: "<<setprecision(3)<<fixed<< function1(a,b);
    }
    else  {
        cout<<"Rezultāts funkcijai Y: "<<setprecision(3)<<fixed<< function2(a,b);
    }
}

void checkValue(float x){
    if (x<=0.0001 || (x+x*x-10)<=0.0001) 
    {
        cout<< "Kļūda! Ar ievadīto X vertibu funkcijas izpildīt nevar. ";
        exit(0);
    }
}

int main()
{
    //cout<<"Jolanta Bērziņa 181RDB178 8. grupa\n";
    float x;
    string answer;
       
    cout<<"Ievadiet vertību X:\n";
    cin >> x;
                
        if (!cin.fail() && (cin.peek()==EOF || cin.peek()=='\n') ) 
        {       
              
            checkValue(x);
            float a=valueA(x);
            float b=valueB(x);
            comparison(a,b);
            
            //iztukšot buferi pirms jaunas datu ievades... 
            cin.clear();
            cin.ignore(100,'\n');
            cout<<"\nVai vēlaties turpināt? (y/n)\n";
            cin>>answer;
            if  (answer=="y"||answer=="Y") main();
        }
        else 
        {
            cout << "\nKļūda! Jūs neievadījāt skaitli!\n";
            cin.clear();
            cin.ignore(100,'\n');
            main();
        }
        
    return 0;    
}


